File: mobile-android/app/src/main/java/com/fitonex/app/data/model/Video.kt
package com.fitonex.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class InstructionVideo(
    val id: String,
    val machine_id: String,
    val uploader_id: String,
    val title: String,
    val description: String? = null,
    val video_key: String,
    val thumb_key: String? = null,
    val duration_sec: Int? = null,
    val created_at: String,
    val like_count: Int = 0,
    val is_liked: Boolean = false,
    val uploader_name: String? = null,
    val machine_name: String? = null
)

@Serializable
data class UploadURLRequest(
    val machine_id: String,
    val title: String,
    val description: String,
    val content_type: String,
    val bytes: Long
)

@Serializable
data class UploadURLResponse(
    val upload_url: String,
    val video_key: String,
    val thumb_upload_url: String? = null,
    val thumb_key: String? = null
)

@Serializable
data class FinalizeVideoRequest(
    val machine_id: String,
    val title: String,
    val description: String,
    val video_key: String,
    val thumb_key: String? = null,
    val duration_sec: Int
)

File: backend/internal/models/video.go
package models

import (
	"time"
)

// InstructionVideo represents an instruction video for a machine
type InstructionVideo struct {
	ID          string    `json:"id" db:"id"`
	MachineID   string    `json:"machine_id" db:"machine_id"`
	UploaderID string    `json:"uploader_id" db:"uploader_id"`
	Title       string    `json:"title" db:"title"`
	Description *string   `json:"description,omitempty" db:"description"`
	VideoKey    string    `json:"video_key" db:"video_key"`
	ThumbKey    *string   `json:"thumb_key,omitempty" db:"thumb_key"`
	DurationSec *int      `json:"duration_sec,omitempty" db:"duration_sec"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	
	// Computed fields
	LikeCount int `json:"like_count"`
	IsLiked   bool `json:"is_liked"`
	PlayURL   string `json:"play_url,omitempty"`
	
	// User info for display
	UploaderName string `json:"uploader_name,omitempty"`
	MachineName  string `json:"machine_name,omitempty"`
}

// VideoLike represents a user's like for a video
type VideoLike struct {
	VideoID   string    `json:"video_id" db:"video_id"`
	UserID    string    `json:"user_id" db:"user_id"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}
